ydb/library/yql/minikql/comp_nodes/ut/mkql_block_exists_ut.cpp

```diff
@@ -0,0 +1,111 @@
+    const auto boolType   = pb.NewDataType(NUdf::TDataType<bool>::Id);
+
+    const auto inputTupleType  = pb.NewTupleType({i32Type, opti32Type, opti32Type, opti32Type});
+    const auto outputTupleType = pb.NewTupleType({i32Type, boolType, boolType, boolType});
+
+    TRuntimeNode::TList input;
+    static_assert(MaxBlockSizeInBytes % 4 == 0);
+
+    for (size_t i = 0; i < 20; i++) {
+        const auto maybeNull = (i % 2) ? pb.NewOptional(pb.NewDataLiteral<i32>(i / 2))
+                                       : pb.NewEmptyOptionalDataLiteral(NUdf::TDataType<i32>::Id);
+
+        const auto inputTuple = pb.NewTuple(inputTupleType, {
+            pb.NewDataLiteral<i32>(i),
+            maybeNull,
+            pb.NewEmptyOptionalDataLiteral(NUdf::TDataType<i32>::Id),
+            pb.NewOptional(pb.NewDataLiteral<i32>(i))
+        });
+
+        input.push_back(inputTuple);
+    }
+
+    const auto list = pb.NewList(inputTupleType, std::move(input));
+
+    auto node = pb.ToFlow(list);
+    node = pb.ExpandMap(node, [&](TRuntimeNode item) -> TRuntimeNode::TList {
+        return {
+            pb.Nth(item, 0),
+            pb.Nth(item, 1),
+            pb.Nth(item, 2),
+            pb.Nth(item, 3)
+        };
+    });
+    node = pb.WideToBlocks(node);
+    if constexpr (Offset > 0) {
+        node = pb.WideSkipBlocks(node, pb.NewDataLiteral<ui64>(Offset));
+    }
+    node = pb.WideMap(node, [&](TRuntimeNode::TList items) -> TRuntimeNode::TList {
+        return {
+            items[0],
+            pb.BlockExists(items[1]),
+            pb.BlockExists(items[2]),
+            pb.BlockExists(items[3]),
+            items[4],
+        };
+    });
+    node = pb.WideFromBlocks(node);
+    node = pb.NarrowMap(node, [&](TRuntimeNode::TList items) -> TRuntimeNode {
+        return pb.NewTuple(outputTupleType, {items[0], items[1], items[2], items[3]});
+    });
+
+    const auto pgmReturn = pb.Collect(node);
+    const auto graph = setup.BuildGraph(pgmReturn);
+    const auto iterator = graph->GetValue().GetListIterator();
+
+    for (size_t i = 0; i < 20; i++) {
+        if (i < Offset) {
+            continue;
+        }
+
+        NUdf::TUnboxedValue outputTuple;
+        UNIT_ASSERT(iterator.Next(outputTuple));
+        const i32 key = outputTuple.GetElement(0).Get<i32>();
+        const bool maybeNull = outputTuple.GetElement(1).Get<bool>();
+        const bool alwaysNull = outputTuple.GetElement(2).Get<bool>();
+        const bool neverNull = outputTuple.GetElement(3).Get<bool>();
+
+        UNIT_ASSERT_VALUES_EQUAL(key, i);
+        UNIT_ASSERT_VALUES_EQUAL(maybeNull, (i % 2) ? true : false);
+        UNIT_ASSERT_VALUES_EQUAL(alwaysNull, false);
+        UNIT_ASSERT_VALUES_EQUAL(neverNull, true);
+    }
+
+    NUdf::TUnboxedValue outputTuple;
+    UNIT_ASSERT(!iterator.Next(outputTuple));
+    UNIT_ASSERT(!iterator.Next(outputTuple));
+}
+
+} //namespace
+
+Y_UNIT_TEST_SUITE(TMiniKQLBlockExistsTest) {
+
+Y_UNIT_TEST(ExistsWithoutOffset) {
```

Comment:
single test is enough with a loop by all possible offsets

Reply:
[Updated](https://github.com/ydb-platform/ydb/pull/3400/files#diff-f9673320d1fb57a4139ff27c7dc5ca7015f9f707599805621544a2a245199328R99-R103).